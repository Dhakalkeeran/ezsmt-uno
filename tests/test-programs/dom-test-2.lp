#theory lp {
    linear_term {
    - : 2, unary;
    * : 1, binary, left;
    + : 0, binary, left;
    - : 0, binary, left
    };

    dom_term {
    - : 3, unary;
    + : 3, unary;
    * : 2, binary, left;
    + : 1, binary, left;
    - : 1, binary, left;
    .. : 0, binary, left
    };

    &dom/0 : dom_term, {=}, linear_term, head;
    &logic/1 : linear_term, head
}.

&logic(lia).
node(2..3).
&dom {(-2-5)..1+(-7); 7+(-1)..(5+2)} = at(X) :- node(X).

