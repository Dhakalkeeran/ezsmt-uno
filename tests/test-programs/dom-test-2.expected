node(3) node(2) 
at(2)=6 at(3)=6 

node(3) node(2) 
at(2)=7 at(3)=6 

node(3) node(2) 
at(2)=7 at(3)=7 

node(3) node(2) 
at(2)=6 at(3)=7 

node(3) node(2) 
at(2)=6 at(3)=(- 6) 

node(3) node(2) 
at(2)=6 at(3)=(- 7) 

node(3) node(2) 
at(2)=7 at(3)=(- 7) 

node(3) node(2) 
at(2)=7 at(3)=(- 6) 

node(3) node(2) 
at(2)=(- 6) at(3)=(- 6) 

node(3) node(2) 
at(2)=(- 7) at(3)=(- 6) 

node(3) node(2) 
at(2)=(- 7) at(3)=(- 7) 

node(3) node(2) 
at(2)=(- 6) at(3)=(- 7) 

node(3) node(2) 
at(2)=(- 6) at(3)=6 

node(3) node(2) 
at(2)=(- 6) at(3)=7 

node(3) node(2) 
at(2)=(- 7) at(3)=7 

node(3) node(2) 
at(2)=(- 7) at(3)=6
